src/renderers/dom/client/ReactEventListener.js

```diff
@@ -172,6 +183,40 @@ var ReactEventListener = {
       TopLevelCallbackBookKeeping.release(bookKeeping);
     }
   },
+  /**
+   * Records the remover for a previously added document event listener.
+   * @param {function} remover A function that removes an added listener.
+   */
+  addDocumentEventListenerRemover: function(remover) {
+    if (remover) {
+      documentEventListenerRemovers.push(remover);
+    }
+  },
+
+  /**
+   * This callback will be invoked when resetDocument is also called.
+   * @param callback
+     */
+  setDocumentResetCallback: function(callback) {
+    documentResetCallback = callback;
+  },
+
+  /**
+   * Removes all previously added document or global event listeners.
+   * @param {object} The document holding the last component.
+   */
+  resetDocument: function(doc) {
+    for (;;) {
+      var remover = documentEventListenerRemovers.pop();
+      if (!remover) {
+        break;
+      }
+      remover();
+    }
+
+    documentResetCallback(doc);
+  },
+
```

Comment:
Unfortunately i had to add these "public" or "visible" apis because other places need to call them.